$MirOS: ports/misc/mc/patches/patch-lib_mc_ext_in,v 1.11 2008/10/16 18:48:13 tg Exp $
--- lib/mc.ext.in.orig	Sat Jul 23 16:50:53 2005
+++ lib/mc.ext.in	Fri Dec 31 17:44:06 2010
@@ -106,8 +106,49 @@
 
 ### Archives ###
 
-# .tgz, .tpz, .tar.gz, .tar.z, .tar.Z
-regex/\.t([gp]?z|ar\.g?[zZ])$
+## CPIO
+
+# .cgz (LZW/UCB or zlib/deflate compressed)
+regex/\.c(pio\.g?[Zz]|gz)$
+	Open=%cd %p#ucpio
+	View=%view{ascii} gzip -dc %f 2>/dev/null | cpio -itv 2>/dev/null
+
+# .cbz (bzip2 compressed)
+regex/\.c(pio\.bz2|bz)$
+	Open=%cd %p#ucpio
+	View=%view{ascii} bzip2 -dc %f 2>/dev/null | cpio -itv 2>/dev/null
+
+# .clz (LZMA compressed)
+regex/\.c(pio\.lzma|lz)$
+	Open=%cd %p#ucpio
+	View=%view{ascii} lzmadec <%f 2>/dev/null | cpio -itv 2>/dev/null
+
+# .cxz (LZMA2 compressed)
+regex/\.c(pio\.xz|xz)$
+	Open=%cd %p#ucpio
+	View=%view{ascii} xzdec <%f 2>/dev/null | cpio -itv 2>/dev/null
+
+## CPIO (newc) - MirOS extension
+
+# .ngz
+shell/.ngz
+	Open=%cd %p#ucpio
+	View=%view{ascii} gzip -dc %f 2>/dev/null | cpio -itv 2>/dev/null
+
+# .nbz
+shell/.nbz
+	Open=%cd %p#ucpio
+	View=%view{ascii} bzip2 -dc %f 2>/dev/null | cpio -itv 2>/dev/null
+
+# .mcz
+shell/.mcz
+	Open=%cd %p#ucpio
+	View=%view{ascii} gzip -dc %f 2>/dev/null | cpio -itv 2>/dev/null
+
+## USTAR
+
+# .tgz, .taz, .tpz, .tar.gz, .tar.z, .tar.Z
+regex/\.t([agp]?z|ar\.g?[zZ])$
 	Open=%cd %p#utar
 	View=%view{ascii} gzip -dc %f 2>/dev/null | tar tvvf -
 
@@ -115,10 +156,21 @@ regex/\.tar\.bz$
 	# Open=%cd %p#utar
 	View=%view{ascii} bzip -dc %f 2>/dev/null | tar tvvf -
 
+# .tar.bz2, .tbz, .tb2
 regex/\.t(ar\.bz2|bz|b2)$
 	Open=%cd %p#utar
 	View=%view{ascii} bzip2 -dc %f 2>/dev/null | tar tvvf -
 
+# .tar.lzma, .tlz
+regex/\.t(ar\.lzma|lz)$
+	Open=%cd %p#utar
+	View=%view{ascii} lzmadec <%f 2>/dev/null | tar tvvf -
+
+# .tar.xz, .txz
+regex/\.t(ar\.xz|xz)$
+	Open=%cd %p#utar
+	View=%view{ascii} xzdec <%f 2>/dev/null | tar tvvf -
+
 # .tar.F - used in QNX
 regex/\.tar\.F$
 	# Open=%cd %p#utar
@@ -135,6 +187,10 @@ regex/\.(tar|TAR)$
 	View=%view{ascii} tar tvvf %f
 
 # lha
+regex/\.[Ll]([Zz][Hh]|[Hh][Aa])$
+	Open=%cd %p#ulha
+	View=%view{ascii} lha l %f
+
 type/^LHa\ .*archive
 	Open=%cd %p#ulha
 	View=%view{ascii} lha l %f
@@ -155,14 +211,6 @@ regex/\.[rR]([aA][rR]|[0-9][0-9])$
 	View=%view{ascii} rar v -c- %f
 
 # cpio
-shell/.cpio.Z
-	Open=%cd %p#ucpio
-	View=%view{ascii} gzip -dc '%f' | cpio -itv 2>/dev/null
-
-shell/.cpio.gz
-	Open=%cd %p#ucpio
-	View=%view{ascii} gzip -dc '%f' | cpio -itv 2>/dev/null
-
 shell/.cpio
 	Open=%cd %p#ucpio
 	View=%view{ascii} cpio -itv <'%f' 2>/dev/null
@@ -191,13 +239,20 @@ regex/\.(rpm|spm)$
 	Open=%cd %p#rpm
 	View=%view{ascii} if rpm --nosignature --version >/dev/null 2>&1; then RPM="rpm --nosignature" ; else RPM="rpm" ; fi ; $RPM -qivlp --scripts %f
 
-# deb
+# deb - Debian packages
 regex/\.u?deb$
-	Open=%cd %p#deb
-	View=%view{ascii} dpkg-deb -c %f
+#	Open=%cd %p#deb
+#	View=%view{ascii} dpkg-deb -c %f
+	Open=%cd %p#uar
+	View=%view{ascii} ar p %f control.tar.gz | gzip -d | tar xOf - ./control; echo; ar p %f data.tar.gz | gzip -d | tar tvvf -
 
+# ipk - like deb, but tgz instead of ar
+shell/.ipk
+	Open=%cd %p#utar
+	View=%view{ascii} gzip -dc %f 2>&- | tar xOf - ./control.tar.gz | gzip -d | tar xOf - ./control; echo; gzip -dc %f 2>&- | tar xOf - ./data.tar.gz | gzip -d | tar tvvf -
+
 # ISO9660
-regex/\.iso$
+regex/\.[Ii][Ss][Oo]$
 	Open=%cd %p#iso9660
 	View=%view{ascii} isoinfo -l -i %f
 
@@ -251,7 +306,7 @@ shell/.info
 regex/\.(so|so\.[0-9\.]*)$
 	View=%view{ascii} file %f && nm %f
 
-regex/(([^0-9]|^[^\.]*)\.([1-9][A-Za-z]*|[ln])|\.man)$
+regex/(([^0-9]|^[^\.]*)\.([1-9][A-Za-z]*|n)|\.man)$
 	Open=case %d/%f in */log/*|*/logs/*) cat %f ;; *) nroff @MAN_FLAGS@ @MANDOC@ %f ;; esac | %var{PAGER:more}
 	View=%view{ascii,nroff} case %d/%f in */log/*|*/logs/*) cat %f ;; *) nroff @MAN_FLAGS@ @MANDOC@ %f ;; esac
 
@@ -271,19 +326,27 @@ shell/.ms
 	View=%view{ascii,nroff} nroff @MAN_FLAGS@ -ms %f
 
 # Manual page - compressed
-regex/([^0-9]|^[^\.]*)\.([1-9][A-Za-z]*|[ln])\.g?[Zz]$
+regex/([^0-9]|^[^\.]*)\.([1-9][A-Za-z]*|n)\.g?[Zz]$
 	Open=case %d/%f in */log/*|*/logs/*) gzip -dc %f ;; *) gzip -dc %f | nroff @MAN_FLAGS@ @MANDOC@ ;; esac | %var{PAGER:more}
 	View=%view{ascii,nroff} case %d/%f in */log/*|*/logs/*) gzip -dc %f ;; *) gzip -dc %f | nroff @MAN_FLAGS@ @MANDOC@ ;; esac
 
-regex/([^0-9]|^[^\.]*)\.([1-9][A-Za-z]*|[ln])\.bz$
+regex/([^0-9]|^[^\.]*)\.([1-9][A-Za-z]*|n)\.bz$
 	Open=case %d/%f in */log/*|*/logs/*) bzip -dc %f ;; *) bzip -dc %f | nroff @MAN_FLAGS@ @MANDOC@ ;; esac | %var{PAGER:more}
 	View=%view{ascii,nroff} case %d/%f in */log/*|*/logs/*) bzip -dc %f ;; *) bzip -dc %f | nroff @MAN_FLAGS@ @MANDOC@ ;; esac
 
-regex/([^0-9]|^[^\.]*)\.([1-9][A-Za-z]*|[ln])\.bz2$
+regex/([^0-9]|^[^\.]*)\.([1-9][A-Za-z]*|n)\.bz2$
 	Open=case %d/%f in */log/*|*/logs/*) bzip2 -dc %f ;; *) bzip2 -dc %f | nroff @MAN_FLAGS@ @MANDOC@ ;; esac | %var{PAGER:more}
 	View=%view{ascii,nroff} case %d/%f in */log/*|*/logs/*) bzip2 -dc %f ;; *) bzip2 -dc %f | nroff @MAN_FLAGS@ @MANDOC@ ;; esac
 
+regex/([^0-9]|^[^\.]*)\.([1-9][A-Za-z]*|[ln])\.lzma$
+	Open=case %d/%f in */log/*|*/logs/*) lzmadec <%f ;; *) lzmadec <%f | nroff @MAN_FLAGS@ @MANDOC@ ;; esac | %var{PAGER:more}
+	View=%view{ascii,nroff} case %d/%f in */log/*|*/logs/*) lzmadec <%f ;; *) lzmadec <%f | nroff @MAN_FLAGS@ @MANDOC@ ;; esac
 
+regex/([^0-9]|^[^\.]*)\.([1-9][A-Za-z]*|[ln])\.xz$
+	Open=case %d/%f in */log/*|*/logs/*) xzdec <%f ;; *) xzdec <%f | nroff @MAN_FLAGS@ @MANDOC@ ;; esac | %var{PAGER:more}
+	View=%view{ascii,nroff} case %d/%f in */log/*|*/logs/*) xzdec <%f ;; *) xzdec <%f | nroff @MAN_FLAGS@ @MANDOC@ ;; esac
+
+
 ### Images ###
 
 type/^GIF
@@ -417,7 +480,7 @@ type/^PDF
 
 # html
 regex/\.([hH][tT][mM][lL]?)$
-	Open=(if test -n "@X11_WWW@" && test -n "$DISPLAY"; then (@X11_WWW@ file://%d/%p &) 1>&2; else links %f || lynx -force_html %f || ${PAGER:-more} %f; fi) 2>/dev/null
+	Open=lynx -force_html %f
 	View=%view{ascii} lynx -dump -force_html %f
 
 # StarOffice 5.2
@@ -526,6 +589,16 @@ type/^bzip
 type/^compress
 	Open=gzip -dc %f | %var{PAGER:more}
 	View=%view{ascii} gzip -dc %f 2>/dev/null
+
+# lzma
+regex/\.lzma$
+	Open=lzmadec <%f | %var{PAGER:more}
+	View=%view{ascii} lzmadec <%f 2>/dev/null
+
+# xz
+regex/\.xz$
+	Open=xzdec <%f | %var{PAGER:more}
+	View=%view{ascii} xzdec <%f 2>/dev/null
 
 
 ### Default ###
